server/odc-service/src/main/java/com/oceanbase/odc/service/iam/UserPermissionService.java

```diff
@@ -128,25 +129,35 @@ public void bindUserAndConnectionAccessPermission(@NonNull Long userId, Long con
     public void bindUserAndDataSourcePermission(@NonNull Long userId, @NonNull Long organizationId,
             @NonNull Long dataSourceId, @NotEmpty List<String> actions) {
         actions.stream().filter(action -> StringUtils.notEquals(action, "create")).distinct().forEach(action -> {
-            PermissionEntity permission = new PermissionEntity();
-            permission.setAction(action);
-            permission.setType(PermissionType.PUBLIC_RESOURCE);
-            permission.setResourceIdentifier("ODC_CONNECTION:" + dataSourceId);
-            permission.setOrganizationId(organizationId);
-            permission.setCreatorId(userId);
-            permission.setBuiltIn(false);
-            permission.setExpireTime(TimeUtils.getMySQLMaxDatetime());
-            permission.setAuthorizationType(AuthorizationType.USER_AUTHORIZATION);
-            PermissionEntity saved = permissionRepository.saveAndFlush(permission);
-
-            UserPermissionEntity userPermission = new UserPermissionEntity();
-            userPermission.setUserId(userId);
-            userPermission.setPermissionId(saved.getId());
-            userPermission.setCreatorId(userId);
-            userPermission.setOrganizationId(organizationId);
-            userPermissionRepository.saveAndFlush(userPermission);
+            getPermission(userId, organizationId, "ODC_CONNECTION:" + dataSourceId, action);
         });
     }
 
+    @SkipAuthorize("internal usage")
+    @Transactional(rollbackFor = Exception.class)
+    public void bindUserAndDataSourceAllPermission(@NonNull Long userId, @NonNull Long organizationId) {
```

Comment:
- `@Transactional` will not take effect
- better not do database operation in a for iteration, use batch save instead
- the method name `getPermission` does not explain its effect

Reply:
All the above three points have been dealt with.